middleware/protect.js

```diff
@@ -31,15 +31,15 @@ function forceLogin (keycloak, request, response, routeName) {
   }
 
   let uuid = UUID();
-  let loginURL = (routeName === "register") ? keycloak.registerUrl(redirectUrl) : keycloak.loginUrl(uuid, redirectUrl);
+  let loginURL = (request.keycloakConfig.action != "undefined" && request.keycloakConfig.action === "/register") ? keycloak.registerUrl(redirectUrl) : keycloak.loginUrl(uuid, redirectUrl);
```

Comment:
action value should be `register` instead of `/register`. use a const defined at the top of the page to contain the action value.